lib/net.js

```diff
@@ -256,6 +258,9 @@ function Socket(options) {
 
   // Used after `.destroy()`
   this[BYTES_READ] = 0;
+
+  this[kLastTimeout] = 0;
+  this[kPrevTimeout] = 0;
```

Comment:
Not necessarily, L394 could have an `if`.
(it's a classical point for polymorphism, but an `if` can accomplish "extendibility"